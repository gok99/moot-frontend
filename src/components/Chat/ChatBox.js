import React, { Component } from 'react';
import { Row, Col } from 'react-bootstrap';

import '../Styles/styles.css';

class ChatBox extends Component {
  render() {
    return (
      <div className="contentbox spacedbox">
        <Row>
          <Col xs={9}>
            <p className="chatboxheader">Anonymous Chat Slot:</p>
          </Col>
          <Col xs={3}>
            <p className="chatempty">EMPTY</p>
            {/* <p className="chatongoing">ONGOING</p> */}
          </Col>
        </Row>
        <hr />
        <p className="chatboxtext">---Anon Chat Info / Post Info---</p>
      </div>
    )
  }
}

export default ChatBox;