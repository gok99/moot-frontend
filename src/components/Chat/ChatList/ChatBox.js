import React from 'react';
import { Row, Col } from 'react-bootstrap';

import '../../Styles/styles.css';
import '../chat.css';

const ChatBox = (props) => {
  const id = props.chatId;
  const status = props.status;
  return (
    <div className="b-chatbox">
      <Row>
        <Col md={8} className="text-chatboxheader">
          <p>Anonymous Chat #{id}:</p>
        </Col>
        <Col md={4} className="text-chatbox">
          { status 
            ? <p className="ongoing">MATCHED!</p>
            : <p className="empty">NOT MATCHED...</p>
          }
        </Col>
      </Row>
    </div>
  );
};

export default ChatBox;