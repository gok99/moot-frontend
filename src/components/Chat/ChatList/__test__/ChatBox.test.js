import React from 'react';
import ReactDOM from 'react-dom';
import ChatBox from '../ChatBox';

import { cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<ChatBox chatId={0} status={true} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
  const tree = renderer.create(<ChatBox chatId={0} status={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});
