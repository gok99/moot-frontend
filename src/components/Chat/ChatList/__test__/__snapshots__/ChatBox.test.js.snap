// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="b-chatbox"
>
  <div
    className="row"
  >
    <div
      className="text-chatboxheader col-md-8"
    >
      <p>
        Anonymous Chat #
        0
        :
      </p>
    </div>
    <div
      className="text-chatbox col-md-4"
    >
      <p
        className="ongoing"
      >
        MATCHED!
      </p>
    </div>
  </div>
</div>
`;
